File: VB/RibbonSample/Converter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Data
Imports System.Globalization
Imports DevExpress.Xpf.Ribbon
Imports System.Windows
Imports DevExpress.Xpf.Ribbon.Themes

Namespace RibbonSample
	Public Class Converter
		Implements IValueConverter
		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
			If value Is Nothing Then
				Return Nothing
			End If
			Dim rb As RibbonPage = CType(value, RibbonPage)
			Dim currentPageindex As Integer = rb.PageCategory.Pages.IndexOf(rb)
			Return If(currentPageindex = 0, New Thickness(PageOffsetHelper.GetPageOffset(rb.Ribbon), 0, 0, 0), CType(rb.FindResource(New RibbonPageThemeKeyExtension() With {.ResourceKey = RibbonPageThemeKeys.CaptionControlMargin}), Thickness))
		End Function

		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
			Return value
		End Function
	End Class
End Namespace

File: CS/RibbonSample/Converter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Data;
using System.Globalization;
using DevExpress.Xpf.Ribbon;
using System.Windows;
using DevExpress.Xpf.Ribbon.Themes;

namespace RibbonSample {
    public class Converter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            if (value == null) return null;
            RibbonPage rb = (RibbonPage)value;
            int currentPageindex = rb.PageCategory.Pages.IndexOf(rb);
            return currentPageindex == 0 ?  new Thickness(PageOffsetHelper.GetPageOffset(rb.Ribbon), 0, 0, 0) : (Thickness)rb.FindResource(new RibbonPageThemeKeyExtension() { ResourceKey = RibbonPageThemeKeys.CaptionControlMargin }) ;
        }

        public object ConvertBack(object value, Type targetType,
            object parameter, CultureInfo culture) {
            return value;
        }
    }
}
